src/main/scala/firrtl/transforms/formal/ConvertAsserts.scala

```diff
@@ -0,0 +1,40 @@
+// See LICENSE for license details.
+
+package firrtl.transforms.formal
+
+import firrtl._
+import firrtl.ir._
+import firrtl.options.Dependency
+
+/**
+  * Convert Asserts
+  *
+  * Replaces all Assert nodes with a gated print-and-stop. This effectively
+  * emulates the assert for IEEE 1364 Verilog.
+  */
+object ConvertAsserts extends Transform with DependencyAPIMigration {
+  override def prerequisites = Nil
+  override def optionalPrerequisites = Nil
+  override def optionalPrerequisiteOf = Seq(
+    Dependency[VerilogEmitter],
+    Dependency[MinimumVerilogEmitter],
+    Dependency[RemoveVerificationStatements])
+
+  override def invalidates(a: Transform): Boolean = false
+
+  def convertAsserts(stmt: Statement): Statement = stmt match {
+    case Verification(Formal.Assert, i, clk, pred, en, msg) =>
+      val nPred = DoPrim(PrimOps.Not, Seq(pred), Nil, pred.tpe)
+      val gatedNPred = DoPrim(PrimOps.And, Seq(nPred, en), Nil, pred.tpe)
+      val stop = Stop(i, 1, clk, gatedNPred)
+      msg match {
+        case StringLit("") => stop
+        case _ => Block(Print(i, msg, Nil, clk, gatedNPred), stop)
+      }
```

Comment:
Should the message be modified to include source locator information similar to Chisel's current output of:

```scala
s"""|Assertion failed: $msg
    |    at $line""".stripMargin
```

Alternatively, maybe it's better to do this in the `chisel3` side?

(And this looks good with @ekiwi's suggestion to ignore the message if it doesn't exist.)

Reply:
I think it might be better to do it on the chisel side, otherwise the messages would be more informative in Verilog than in SystemVerilog

Reply:
That sounds good to me!